docs/GettingStarted.md

```diff
@@ -1617,3 +1620,57 @@ However, additional instrumentation provided by Datadog can be activated alongsi
 | `OpenTracing::FORMAT_TEXT_MAP` | Yes        |                        |
 | `OpenTracing::FORMAT_RACK`     | Yes        | Because of the loss of resolution in the Rack format, please note that baggage items with names containing either upper case characters or `-` will be converted to lower case and `_` in a round-trip respectively. We recommend avoiding these characters, or accommodating accordingly on the receiving end. |
 | `OpenTracing::FORMAT_BINARY`   | No         |                        |
+
+### Debugging
+
+If you are having difficulty using the tracer, the library offers a few debugging features to give you more insight into the internals of tracing.
+
+#### Using debug logging
+
+Debug logging can be activate to produce verbose logs that detail the operations of the tracer, by enabling it on the tracer settings:
+
+```ruby
+# config/initializers/datadog.rb
+require 'ddtrace'
+
+Datadog.configure do |c|
+  # Activates debug mode for the tracer
+  c.tracer debug: true
+end
+```
+
+Both the tracer and its integrations will produce detailed messages of operations to your configured log file.
+
+Be aware that *debug logging can produce many messages*, especially in applications with higher throughput. We generally recommend that this flag only be used in controlled, fail-safe environments (e.g. staging)
+
+#### Using debug stats
+
+To gather metrics regarding the internals of the trace library (such as number of traces flushed, number of errors, etc), one can enable debug metrics, which are collected using the `dogstatsd-ruby` library.
+
+To activate, first add `gem 'dogstatsd-ruby'` to your Gemfile, then add the following to your configuration file:
+
+```ruby
+# config/initializers/datadog.rb
+require 'datadog/statsd'
+require 'ddtrace'
+
+Datadog.configure do |c|
+  # Activates Statsd metric collection for debugging purposes.
+  # Configure with host and port of Datadog agent; defaults to 'localhost:8125'.
+  c.tracer statsd: Datadog::Statsd.new
```

Comment:
We may expand this part saying that users can reuse their statsd client in case it's already initalized in another part of their application. I would say to double check thread-safety in that case.